libcontainerd/client_linux.go

```diff
@@ -86,7 +86,7 @@ func (clnt *client) AddProcess(ctx context.Context, containerID, processFriendly
 		Rlimits:         convertRlimits(sp.Rlimits),
 	}
 
-	iopipe, err := p.openFifos(sp.Terminal)
+	iopipe, err := p.openFifos(sp.Terminal, nil)
```

Comment:
Actually, this case isn't that different. Only difference is that you can't reproduce it with a logdriver(that issue we could fix with easy validation for containers). If the attach fails in here you will still have a process running without docker having control of the io streams.